File: 42578.cpp
#include <string>
#include <vector>
#include <iostream>
#include <unordered_map>
#include <algorithm>

using namespace std;

unordered_map<string, int> clothesMap;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    
    for(auto each : clothes) {
        string category = each[1];
        if(clothesMap.find(category) == clothesMap.end()) {
            clothesMap[category] = 1;
        } else {
            clothesMap[category] += 1;
        }
    }
    
    for(auto each : clothesMap) {
        answer *= (each.second + 1);
    }
    answer -= 1;
    
    
    return answer;
}

File: 42578.swift
import Foundation

var categoryCount: Dictionary<String, Int> = [:]

func solution(_ clothes:[[String]]) -> Int {
    var answer = 1
    clothes.forEach { if let count = categoryCount[$0[1]] { categoryCount[$0[1]] = count + 1 } else  { categoryCount[$0[1]] = 1 } }
    categoryCount.forEach { key, value in answer = answer * (value + 1) }
    
    return answer - 1
}
